In [ ]:
import socket
def get_ip_address(website_url):
    try:
        ip_address = socket.gethostbyname(website_url)
        print(f"The IP address of {website_url} is {ip_address}")
    except socket.gaierror:
        print(f"Unable to get the IP address for {website_url}")
# Example usage
website = input("Enter the website URL (without 'https://'): ")
get_ip_address(website)

![Alt Text](Images/Week1a.jpg)

1) The IP address of www.gold.ac.uk is 159.100.136.66
2) The IP address of www.google.com is 172.217.16.228
3) The IP address of www.youtube.com is 142.250.187.238

In [ ]:
import subprocess
def tracert(domain):
    try:
        result = subprocess.run(["tracert", domain], capture_output=True, text=True)
        print(result.stdout)
    except FileNotFoundError:
        print("tracert command not found. Make sure it's available.")
    except Exception as e: # Catching general exceptions for now
        print(f"An error occurred: {e}")
domain = input("Enter the website or IP address: ")
tracert(domain)

![alt text](Images/Results1.png)

![alt text](Images/Results2.png)

![alternatvie text](Images/results4.jpg)

In [ ]:
import socket

client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

server_address = ('www.gold.ac.uk', 80)

client_socket.connect(server_address)

request = "GET / HTTP/1.1\r\nHost: www.gold.ac.uk\r\n\r\n"
client_socket.send(request.encode())

response = client_socket.recv(4096)
print(response.decode())

client_socket.close()

![alternatvie text](Images/results3.jpg)

In [ ]:
import requests
response = requests.get('http://www.gold.ac.uk')
print(response.text)

![alternatvie text](Images/results5.jpg)

In [3]:
import requests

url = "https://reqres.in/api/users"
data = {
    "name": "John Doe",
    "job": "Developer"
}

response = requests.post(url, json=data)

print(f"Status Code: {response.status_code}")
print("Response Body:", response.json())


Status Code: 201
Response Body: {'name': 'John Doe', 'job': 'Developer', 'id': '696', 'createdAt': '2025-03-03T21:50:38.472Z'}


Status Code: 201
Response Body: {'name': 'John Doe', 'job': 'Developer', 'id': '556', 'createdAt': '2025-02-27T18:28:49.918Z'}

In [4]:
import requests

url = "https://reqres.in/api/users/2" 
data = {
    "name": "John Doe",
    "job": "Senior Developer"
}

response = requests.put(url, json=data, timeout=10)

print(f"Status Code: {response.status_code}")
print("Response Body:", response.json())


Status Code: 200
Response Body: {'name': 'John Doe', 'job': 'Senior Developer', 'updatedAt': '2025-03-03T21:50:54.058Z'}


Status Code: 200
Response Body: {'name': 'John Doe', 'job': 'Senior Developer', 'updatedAt': '2025-02-27T18:35:36.785Z'}

In [5]:
import requests

url = 'https://reqres.in/api/users/2'
response = requests.delete(url)
print(f"Status Code: {response.status_code}")

if response.status_code == 204:
    print("Resource successfully deleted.")
else:
    print("failed to delete")

Status Code: 204
Resource successfully deleted.


Status Code: 204
Resource successfully deleted.

![Alt Text](Images/StatusCodesMeaning.jpg)